### DISCUSSION TITLE
WRC+ Telemetry

### DISCUSSION THREAD

**Discussion by Paulr552** [RallyDataJunkie/rallydatajunkie.github.io#2] [Show and tell]
Hi Tony, Loved reading up on the rally visualisations for telemetry and for stage times. You have done some trojan work. Would you mind sharing how you get the WRC+ Telemetry csv files that you use. Thanks

**Comment by psychemedia** [comment]
You can get low sample rate GPS during a stage with a recipe of the following form:

```python
from sqlite_utils import Database

db = Database("testlive-wrc-sweden-22.db")

url='https://webappsdata.wrc.com/srv/wrc/json/api/liveservice/getData?timeout=500'

import time

shakedown = db["SS14"] 
while True:
    livetimes = requests.get(url)
    df = json_normalize(livetimes.json(), '_entries', meta='timestamp')
    # I save to a db but you could save the data as a file
    shakedown.upsert_all(df.to_dict(orient='records'), pk=['name','utx'])
    time.sleep(2)
```

You have to do a bit of fiddling to decode which drive the `name` column relates to (offhand, I don't remember how you do the lookup for that). 

<img width="661" alt="image" src="https://user-images.githubusercontent.com/82988/156143629-13460185-1248-4907-966f-03d136c6277b.png">

Note that the telemetry in that feed is largely pointless; the vehicle,  timestamp and GPS location are the main features.

The actual telemetry files are published several hours after a loop and I found them by poking around in the plumbing of  WRC Live+ pages, though you can (could?) actually access them without authentication.

<s>Somewhen between Monte Carlo and Sweden, things seem to have changed on the API front and my scrapers have broken... (Just looking at them now.)</s> FIXED (I've been wary of sharing code in case it jeopardised my data supply. <s>But it seems it's jeopardised now anyway... </s>I do have archives of old telemetry though, which could be of historical interest...)
